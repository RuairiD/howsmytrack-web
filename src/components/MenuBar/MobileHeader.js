import React from "react";

import { Badge, Typography } from "antd";
import { Div, Img, Span } from "lemon-reset";

const MobileHeader = ({ mobileMenuCollapsed, notifications }) => (
    <Div style={{ display: "flex" }}>
        <Span style={{
            marginTop: "auto",
            marginBottom: "auto",
        }}
        >
            <Typography.Text>
                how's my track?
            </Typography.Text>
            {mobileMenuCollapsed && (
                <Badge
                    count={notifications}
                    style={{
                        marginLeft: "0.5em",
                    }}
                />
            )}
        </Span>
        <Img
            alt=""
            src="/logo128.png"
            style={{
                marginLeft: "auto",
                width: "32px",
                height: "32px",
            }}
        />
    </Div>
);

export default MobileHeader;
