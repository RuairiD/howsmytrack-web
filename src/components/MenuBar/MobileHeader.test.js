import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import MobileHeader from "./MobileHeader";

describe("MobileHeader", () => {
    it("renders notifications Badge when relevant menu is collapsed", () => {
        const wrapper = shallow(
            <MobileHeader
                mobileMenuCollapsed
                notifications={2}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("does not render notifications Badge when relevant menu is open", () => {
        const wrapper = shallow(
            <MobileHeader
                notifications={2}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
