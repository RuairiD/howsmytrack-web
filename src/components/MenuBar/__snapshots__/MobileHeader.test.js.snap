// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileHeader does not render notifications Badge when relevant menu is open 1`] = `
<Div
  className=""
  style={
    Object {
      "display": "flex",
    }
  }
>
  <Span
    className=""
    style={
      Object {
        "marginBottom": "auto",
        "marginTop": "auto",
      }
    }
  >
    <Text>
      how's my track?
    </Text>
  </Span>
  <Img
    alt=""
    className=""
    src="/logo128.png"
    style={
      Object {
        "height": "32px",
        "marginLeft": "auto",
        "width": "32px",
      }
    }
  />
</Div>
`;

exports[`MobileHeader renders notifications Badge when relevant menu is collapsed 1`] = `
<Div
  className=""
  style={
    Object {
      "display": "flex",
    }
  }
>
  <Span
    className=""
    style={
      Object {
        "marginBottom": "auto",
        "marginTop": "auto",
      }
    }
  >
    <Text>
      how's my track?
    </Text>
    <Badge
      count={2}
      dot={false}
      overflowCount={99}
      showZero={false}
      style={
        Object {
          "marginLeft": "0.5em",
        }
      }
    />
  </Span>
  <Img
    alt=""
    className=""
    src="/logo128.png"
    style={
      Object {
        "height": "32px",
        "marginLeft": "auto",
        "width": "32px",
      }
    }
  />
</Div>
`;
